src/job-exporter/src/collector.py

```diff
@@ -470,20 +473,27 @@ def collect_impl(self):
         return self.collect_container_metrics(stats_obj, gpu_infos, all_conns)
 
     @staticmethod
-    def parse_from_labels(labels):
+    def parse_from_labels(inspect_info, gpu_infos):
         gpu_ids = []
-        other_labels = {}
-
-        for key, val in labels.items():
-            if "container_label_GPU_ID" == key:
-                s2 = val.replace("\"", "").split(",")
-                for id in s2:
-                    if id:
-                        gpu_ids.append(id)
-            else:
-                other_labels[key] = val
-
-        return gpu_ids, other_labels
+        result_labels = {}
+
+        result_labels["username"] = inspect_info.username or "unknown"
+        result_labels["job_name"] = inspect_info.job_name or "unknown"
+        result_labels["role_name"] = inspect_info.role_name or "unknown"
+        result_labels["task_index"] = inspect_info.task_index or "unknown"
+
+        if inspect_info.gpu_ids:
+            ids = inspect_info.gpu_ids.replace("\"", "").split(",")
+            for id in ids:
+                if id.isdigit():
```

Comment:
In which case, it is directlt the gpu id, and in which case it is just a UUID?

Reply:
added comment in code.

Containers from yarn will have minor in gpu id, which are digits.

Containers from kube launcher will have UUID in gpu id.